### ISSUE TITLE
ShapeNet-55

### ISSUE THREAD

**Issue by Phoebe-ovo** [lulutang0608/Point-BERT#32]
Why not use the original ShapeNetCore dataset directly?
And how did you generate ShapeNet-55 from ShapeNetCore?

**Comment by yuxumin** [comment]
Hi, thanks for your interest in our paper.
The complete point clouds in ShapeNet-55 are the same with ShapeNetCore. So, in fact, we use the original ShapeNetCore for pretraining.
(ShapeNet-55 is a dataset for point cloud completion. But the data for these two dataset are just the same. We claim we use ShapeNetCore in our paper. Use it in our repo just due to our repo are partially based on PoinTr, which proposes ShapeNet-55)

Hi, thanks for your interest in our paper.
The complete point cloud in ShapeNet-55 is the same as in ShapeNetCore. So, in fact, we use the original ShapeNetCore for pre-training.
(ShapeNet-55 is the dataset used for point cloud completion. But the data for both datasets is the same. We claim that we used ShapeNetCore in our paper. We use it in our repo only because our repo is  based on PoinTr, which proposes ShapeNet-55)

**Comment by Phoebe-ovo** [comment]
Thank you for your quick reply~
What's the code for generating ShapeNet-55?

**Comment by yuxumin** [comment]
In fact, they are the same in term of data.

**Comment by Phoebe-ovo** [comment]
Given the original 3D model in the shapenet dataset(.obj files), I want to know how did you get the groundtruth point cloud?

**Comment by yuxumin** [comment]
We use the data from ShapeNetCore in the training of dVAE and MPM tasks for Transformers.
- In dVAE: The model (dVAE) is required to reconstruct the input point clouds. In that case, the groundtruth point cloud is exactly itself.
- In MPM tasks: We randomly mask some local patches in point cloud from ShapeNetCore and require the Transformers to predict the missing local patches by discrete tokens. The groundtruth discrete tokens are generated by infering the missing local patches with the *Tokenizer* in the pretrained dVAE.

**Comment by yuxumin** [comment]
If I misunderstood the question you want to ask, plz feel free to continue the question.

**Comment by Phoebe-ovo** [comment]
ShapeNetCore seems don't provide point cloud directly, it provides the mesh and normal vector.
I want to ask how to generate the point cloud from the mesh and normal vector?

**Comment by yuxumin** [comment]
Sorry for misunderstanding  your question.
The point clouds are obtained by random-sampling or other sampling methods. An example code is provided in [AtlasNet/OCNN](https://github.com/ThibaultGROUEIX/AtlasNet/tree/master/auxiliary/sampling_and_meshing)

**Comment by Phoebe-ovo** [comment]
Thank you for your patient reply. I will try it by myself.

**Comment by yuxumin** [comment]
Close it since no response. Feel free to re-open it if problems still exist
